modules/data-streams/src/test/java/org/elasticsearch/datastreams/UpdateTimeSeriesRangeServiceTests.java

```diff
@@ -140,23 +140,7 @@ public void testUpdateTimeSeriesTemporalRange_NoUpdateBecauseReplicated() {
         ).getMetadata();
         DataStream d = metadata.dataStreams().get(dataStreamName);
         metadata = Metadata.builder(metadata)
-            .put(
-                new DataStream(
-                    d.getName(),
-                    d.getIndices(),
-                    d.getGeneration(),
-                    d.getMetadata(),
-                    d.isHidden(),
-                    true,
-                    d.isSystem(),
-                    d.isAllowCustomRouting(),
-                    d.getIndexMode(),
-                    d.getLifecycle(),
-                    d.isFailureStore(),
-                    d.getFailureIndices(),
-                    null
-                )
-            )
+            .put(new DataStream.Builder(d).setReplicated(true).setRolloverOnWrite(false).setAutoShardingEvent(null).build())
```

Comment:
In most places, I added `setRolloverOnWrite(false)`, because that's what the original constructor did there. I'm guessing it doesn't actually need to be set in a lot of those places, though. What are people's opinion on (me) taking some time to go through these changes and check on a per-case basis whether it actually needs to be set explicitly?

Reply:
I have a general concern that because the setting was being set under all these different constructors that maybe there are some cases where we're unintentionally clearing the flag. I pointed one out that I was perplexed by in my review. Maybe @gmarouli can weigh in on this?

Reply:
You are absolutely right, this is definitely a bug, and I need to look into this, it should not clear the flag unless it was rolled over.

Reply:
We can split the PRs, I can work on it after this is merged.

Reply:
Actually, I will fix it first so I can backport it easier to 8.13

Reply:
See https://github.com/elastic/elasticsearch/pull/107122 for a fix.